File: WickedTuna/WickedTunaCore/Boats/Boat.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;
using WickedTunaCore.Common;
using WickedTunaCore.Users;

namespace WickedTunaCore.Boats
{
    public class Boat
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public int NumberOfEngines { get; set; }
        public float EnginePower { get; set; }
        public float MaximumSpeed { get; set; }
        public Address Address { get; set; }
        public string Description { get; set; }
        public int Capacity { get; set; }
        public float CancellationFee { get; set; }

        [JsonIgnore]
        public BoatOwner BoatOwner { get; set; }
        public string BoatOwnerId { get; set; }
        public ICollection<BoatAvailablePeriod> BoatAvailablePeriods { get; set; }
        [JsonIgnore]
        public ICollection<BoatReservation> BoatReservations { get; set; }

    }
}

File: frontend/src/app/_models/boat.ts
import { Address } from "./address";
import { AvailablePeriod } from "./availablePeriod";
import { BoatOption } from "./boatOptions";

export class Boat {
    id?: string;
    name = '';
    type = '';
    numberOfEngines = 0;
    enginePower = 0;
    maximumSpeed = 0;
    address = new Address();
    capacity = 0;
    cancellationFee = 0;
    boatAvailablePeriods! : AvailablePeriod[];
    description = '';
    price = 0;
    boatAdditionalOptions! : BoatOption[];


    constructor() {}
}